### PR TITLE
Merge Master to 3.0.7

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs [hunk 1]
```diff
@@ -525,11 +525,11 @@ public void CoinstakeAge_PrevOutIsCoinstake_BeforeActivation_Mainnet()
         [Fact]
         public void CoinstakeAge_PrevOutIsCoinstake_AfterActivation_Mainnet()
         {
-            int activationHeight = PosConsensusOptions.CoinstakeMinConfirmationActivationHeightMainnet;
-            int afterActivationHeight = activationHeight + 1000;
-
-            Assert.True(this.WasUtxoSelectedForStaking(KnownNetworks.StratisMain, afterActivationHeight, afterActivationHeight - 498, true));
-            Assert.False(this.WasUtxoSelectedForStaking(KnownNetworks.StratisMain, afterActivationHeight, afterActivationHeight - 497, true));
+            // The logic here is that, before the activation, a coinstake UTXO requires 50 confirmations on mainnet to be used as a staking candidate.
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs]: This test looks screwy, like it's been swapped around with one of the others. I'll check.

File: src/Stratis.Bitcoin.FullNode.sln [hunk 2]
```diff
@@ -335,7 +335,6 @@ Global
 		{246B2AEE-EC8B-4BCA-A6AC-959F5651AE4F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{246B2AEE-EC8B-4BCA-A6AC-959F5651AE4F}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{246B2AEE-EC8B-4BCA-A6AC-959F5651AE4F}.Release|Any CPU.ActiveCfg = Release|Any CPU
```
Comment (ArcSin2000X) [inline src/Stratis.Bitcoin.FullNode.sln]: is that intentional to remove it here?

**Comment by quantumagi** [comment]
#4051 has been merged by #4066. No need to merge it again

File: src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs]: @zeptin in 307 it looks like this?

https://github.com/stratisproject/StratisBitcoinFullNode/blob/884dbae65c46a48db9da5a82d758eecf106f7584/src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs]: Will fix

File: src/Stratis.Bitcoin.FullNode.sln (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.FullNode.sln]: Fixed

**Pull request merged**